### PR TITLE
chore: improve jsdoc of lib/core/constants.js

### PR DESCRIPTION
**Pull request by Uzlopak**
image

### CODE REVIEW

**Comment by tsctx** [comment]
Strict typings aren't required as only tree construction is used.

**Comment by Uzlopak** [comment]
@tsctx 

Wasnt there an issue, where we target to generate the typings from the jsdoc documentation? For that these changes are created.

**Comment by Uzlopak** [comment]
@tsctx 
https://github.com/nodejs/undici/issues/3132#issuecomment-2303700657

File: lib/core/constants.js [hunk 1]
```diff
@@ -100,7 +99,13 @@ const wellknownHeaderNames = [
   'X-Powered-By',
   'X-Requested-With',
   'X-XSS-Protection'
-]
+])
+
+/** @type {Record<typeof wellknownHeaderNames[number]|Lowercase<typeof wellknownHeaderNames[number]>, string>} */
```
Comment (tsctx) [inline lib/core/constants.js:104]:
```suggestion
/** @type {Record<string, string>} */
```

**Review by tsctx — APPROVED** [review]

**Pull request merged**